src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -0,0 +1,270 @@
+        public ServiceAccount? ServiceAccount { get; set; }
+        string? IWinSWConfiguration.ServiceAccountPassword => ServiceAccount != null ? ServiceAccount.Password : null;
+        string? IWinSWConfiguration.ServiceAccountUser => ServiceAccount != null ? ServiceAccount.User : null;
+        bool IWinSWConfiguration.AllowServiceAcountLogonRight { get; }
+
+
+        [YamlMember(Alias = "log")]
+        public YAMLLog? _YAMLLog { get; set; }
+        public Log? Log => _YAMLLog;
+
+        string IWinSWConfiguration.LogDirectory => Log != null ? Log.Directory : null;
+        string IWinSWConfiguration.LogMode => Log != null ? Log.Directory : null;
+
+        
+        [YamlMember(Alias = "download")]
+        public List<Download> Downloads { get; set; }
+
+        [YamlMember(Alias = "caption")]
+        public string Caption { get; set; }
+
+        
+
+        [YamlMember(Alias = "arguments")]
+        public string? Arguments
+        {
+            get => GetArguments(Arguments, ArgType.arg);
+            set { }
+        }
+
+        [YamlMember(Alias = "startArguments")]
+        public string? StartArguments {
+            get => GetArguments(StartArguments, ArgType.startarg);
+            set { } 
+        }
+
+        [YamlMember(Alias = "stopArguments")]
+        public string? StopArguments {
+            get => GetArguments(StopArguments, ArgType.stoparg);
+            set { }
+        }
+
+        [YamlMember(Alias = "stopExecutable")]
+        public string? StopExecutable { get; set; }
+
+        
+
+        [YamlMember(Alias = "stopParentProcessFirst")]
+        public bool StopParentProcessFirst { get; set; }
+        
+        [YamlMember(Alias = "resetFailureAfter")]
+        public TimeSpan ResetFailureAfter { get; set; }
+
+        [YamlMember(Alias = "stopTimeout")]
+        public TimeSpan StopTimeout { get; set; }
+        
+        [YamlMember(Alias = "startMode")]
+        public StartMode StartMode { get; set; }
+
+        [YamlMember(Alias = "serviceDependencies")]
+        public string[] ServiceDependencies { get; set; }
+
+        [YamlMember(Alias = "waitHint")]
+        public TimeSpan WaitHint { get; set; }
+
+        [YamlMember(Alias = "sleepTime")]
+        public TimeSpan SleepTime { get; set; }
+
+        [YamlMember(Alias = "interactive")]
+        public bool Interactive { get; set; }
+
+        [YamlMember(Alias = "priority")]
+        public ProcessPriorityClass Priority { get; set; }
+
+        [YamlMember(Alias = "beepOnShutdown")]
+        public bool BeepOnShutdown { get; set; }
+
+        [YamlMember(Alias = "environmentVariables")]
+        public Dictionary<string, string> EnvironmentVariables { get; set; }
+
+        [YamlMember(Alias = "failureActions")]
+        public List<YAML_SC_ACTION> YamlFailureActions { get; set; }
+
+        [YamlMember(Alias = "delayedAutoStart")]
+        public bool DelayedAutoStart { get; set; }
+
+        public class YAML_SC_ACTION
```

Comment:
Would be great to rename the class so that it does not look like a static variable. E.g. `YamlFailureAction` or so. In the future itcan be also generalized as we discussed in the chat

Reply:
Updated.